cmd/query/app/server.go

```diff
@@ -62,18 +67,27 @@ func NewServer(logger *zap.Logger, querySvc *querysvc.QueryService, options *Que
 		return nil, err
 	}
 
+	if (options.TLSHTTP.Enabled || options.TLSGRPC.Enabled) && (grpcPort == httpPort) {
+		return nil, errSharedTLSPort
```

Comment:
errSharedTLSPort is not reused or needs to be pre-allocated for performance. Please inline it.